Add web worker to docs (#7442)
software-mansion/react-native-reanimated: packages/docs-reanimated/docs/threading/webWorker.mdx

---
sidebar_position: 5
---

# WebWorker

> Web Workers makes it possible to run a worklet function in a separate/background thread (`WorkletRuntime`).

Reviewer: ```suggestion
Web Workers makes it possible to run JavaScript code on a separate background thread.
```

> Once created, a worker instance allows communication between the main UI thread and the background thread using message passing.

Reviewer: ```suggestion
Once created, a worker instance allows communication between the [React Native runtime](/docs/next/fundamentals/glossary#javascript-thread) and the background thread using message passing.
```

:::

## Reference

### Usage in JavaScript

```jsx
import { WebWorker } from 'react-native-worklets';

// Create a new worker, name has to be unique
const worker = new WebWorker('WebWorker#1', () => {
  'worklet';
  console.log('Hello from the background thread');

  // Define the handler for messages received FROM the main thread
  global.onmessage = (event) => {
    console.log(`Worker received: ${event.data}`);

    // Example: Send a message back to the main thread
    const result = event.data * 2;
    global.postMessage(`Worker calculated: ${result}`);
  };
});

// Define the handler for messages received FROM the worker thread
worker.onmessage = (event) => {
  console.log(`Main thread received: ${event.data}`);
};

// Define the handler for errors occurring IN the worker thread
worker.onerror = (event) => {
  console.error(`Error in worker: ${event.data}`);
};

// Send a message TO the worker thread
worker.postMessage(10);

// Later, when the worker is no longer needed:

> // worker.terminate();

Reviewer: ```suggestion
worker.terminate();
```
